packages/cli/src/commands/export-dynamic-plugin/command.ts

```diff
@@ -74,5 +75,50 @@ export async function command(opts: OptionValues): Promise<void> {
     });
   }
 
+  await checkBackstageSupportedVersions(targetPath);
+
   await applyDevOptions(opts, rawPkg.name, roleInfo, targetPath);
 }
+
+async function checkBackstageSupportedVersions(targetPath: string) {
+  const targetPackageFile = path.join(targetPath, 'package.json');
+  const targetPackage = await fs.readJSON(targetPackageFile);
+  const supportedVersions: string | undefined =
+    targetPackage.backstage?.['supported-versions'];
+  const backstageJson = path.join(paths.targetRoot, '/backstage.json');
+  if (fs.existsSync(backstageJson)) {
```

Comment:
would it better to reduce nesting and just return at this point if it doesn't exist?

Reply:
You're right.
I couldn't do it before, before the code wasn't extracted as a function.
I'll do it now.

Reply:
Fixed it, and rebased the code